src/task_list_types.F

```diff
@@ -21,20 +21,36 @@ MODULE task_list_types
 
    CHARACTER(len=*), PARAMETER, PRIVATE :: moduleN = 'task_list_types'
 
-! **************************************************************************************************
+!>        tasks(1,i) := destination
+!>        tasks(2,i) := source
+!>        tasks(3,i) := compressed type (iatom, jatom, ....)
+!>        tasks(4,i) := type (0: replicated, 1: distributed local, 2: distributed generalised)
+!>        tasks(5,i) := cost
+!>        tasks(6,i) := alternate destination code (0 if none available)
+
+   TYPE task_type
+      INTEGER(kind=int_8)                           :: destination ! index 1
+      INTEGER(kind=int_8)                           :: source ! index 2
+      INTEGER(kind=int_8)                           :: pair ! index 3
+      INTEGER(kind=int_8)                           :: TYPE ! index 4
+      INTEGER(kind=int_8)                           :: cost ! index 5
+      INTEGER(kind=int_8)                           :: alt_destination ! index 6
+   END TYPE task_type
+
    TYPE task_list_type
-      INTEGER(kind=int_8), DIMENSION(:, :), POINTER :: tasks
+      TYPE(task_type), DIMENSION(:), POINTER        :: tasks
```

Comment:
is there a chance we can move to an `allocatable` her? This would simplify the realloc via `move_alloc`?